<Type Name="InvalidPipelineStoreException" FullName="System.AddIn.Hosting.InvalidPipelineStoreException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="9ee171081598fd2f1676d309468e78e4022b5e71" /><Meta Name="ms.sourcegitcommit" Value="11d168140aa8fade0768c2a9dde3e3bcacfdfb7d" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="ja-JP" /><Meta Name="ms.lasthandoff" Value="05/27/2020" /><Meta Name="ms.locfileid" Value="83956562" /></Metadata><TypeSignature Language="C#" Value="public class InvalidPipelineStoreException : Exception" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit InvalidPipelineStoreException extends System.Exception" />
  <TypeSignature Language="DocId" Value="T:System.AddIn.Hosting.InvalidPipelineStoreException" />
  <TypeSignature Language="VB.NET" Value="Public Class InvalidPipelineStoreException&#xA;Inherits Exception" />
  <TypeSignature Language="C++ CLI" Value="public ref class InvalidPipelineStoreException : Exception" />
  <TypeSignature Language="F#" Value="type InvalidPipelineStoreException = class&#xA;    inherit Exception" />
  <AssemblyInfo>
    <AssemblyName>System.AddIn</AssemblyName>
    <AssemblyVersion>3.5.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="94343-101">ディレクトリが見つからず、パイプライン ルート パスまたはアドイン パスにアクセスするためのアクセス許可がユーザーにない場合にスローされる例外です。</span><span class="sxs-lookup"><span data-stu-id="94343-101">The exception that is thrown when a directory is not found and the user does not have permission to access the pipeline root path or an add-in path.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="94343-102"><xref:System.IO.DirectoryNotFoundException> とは異なり、この例外ではパス情報が提供されません。</span><span class="sxs-lookup"><span data-stu-id="94343-102">Unlike <xref:System.IO.DirectoryNotFoundException>, this exception does not provide any path information.</span></span> <span data-ttu-id="94343-103">これにより、悪意のあるユーザーが間違ったパスを意図的に指定し、実際のパスを判断するために例外によって返される情報を利用することはできなくなります。</span><span class="sxs-lookup"><span data-stu-id="94343-103">This prevents a malicious user from deliberately specifying erroneous paths and using the information returned by the exception to determine actual paths.</span></span>  
  
 <span data-ttu-id="94343-104">この例外は、アドインとパイプライン情報のストア (<xref:System.AddIn.Hosting.AddInStore.FindAddIns%2A>、<xref:System.AddIn.Hosting.AddInStore.Rebuild%2A>、<xref:System.AddIn.Hosting.AddInStore.RebuildAddIns%2A>、<xref:System.AddIn.Hosting.AddInStore.Update%2A>、および <xref:System.AddIn.Hosting.AddInStore.UpdateAddIns%2A>の作成と更新を行う、次の探索方法によってスローされます。</span><span class="sxs-lookup"><span data-stu-id="94343-104">This exception is thrown by the following discovery methods that build and update the store of add-in and pipeline information: <xref:System.AddIn.Hosting.AddInStore.FindAddIns%2A>,<xref:System.AddIn.Hosting.AddInStore.Rebuild%2A>, <xref:System.AddIn.Hosting.AddInStore.RebuildAddIns%2A>, <xref:System.AddIn.Hosting.AddInStore.Update%2A>, and <xref:System.AddIn.Hosting.AddInStore.UpdateAddIns%2A>.</span></span>  
  
 ]]></format>
    </remarks>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.AddIn</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="94343-105"><see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="94343-105">Initializes a new instance of the <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public InvalidPipelineStoreException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AddIn.Hosting.InvalidPipelineStoreException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; InvalidPipelineStoreException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.AddIn</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="94343-106"><see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="94343-106">Initializes a new instance of the <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> class.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public InvalidPipelineStoreException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AddIn.Hosting.InvalidPipelineStoreException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; InvalidPipelineStoreException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.AddIn.Hosting.InvalidPipelineStoreException : string -&gt; System.AddIn.Hosting.InvalidPipelineStoreException" Usage="new System.AddIn.Hosting.InvalidPipelineStoreException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.AddIn</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="94343-107">エラーの説明。</span><span class="sxs-lookup"><span data-stu-id="94343-107">A description of the error.</span></span></param>
        <summary><span data-ttu-id="94343-108">指定されたメッセージを使用して <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="94343-108">Initializes a new instance of the <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> class with the specified message.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected InvalidPipelineStoreException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AddIn.Hosting.InvalidPipelineStoreException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; InvalidPipelineStoreException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new System.AddIn.Hosting.InvalidPipelineStoreException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; System.AddIn.Hosting.InvalidPipelineStoreException" Usage="new System.AddIn.Hosting.InvalidPipelineStoreException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.AddIn</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="94343-109">シリアル化されたオブジェクト データです。</span><span class="sxs-lookup"><span data-stu-id="94343-109">The serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="94343-110">転送元または転送先のオブジェクト データに関する文脈情報。</span><span class="sxs-lookup"><span data-stu-id="94343-110">The contextual information about the source or destination object data.</span></span></param>
        <summary><span data-ttu-id="94343-111">シリアル化情報とストリーム コンテキスト情報を使用して、<see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="94343-111">Initializes a new instance of the <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> class with serialization and streaming context information.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="94343-112">このコンストラクターは、逆シリアル化中に、ストリームで送信される例外オブジェクトを再構築するために呼び出されます。</span><span class="sxs-lookup"><span data-stu-id="94343-112">This constructor is called during deserialization to reconstitute an exception object that is transmitted over a stream.</span></span> <span data-ttu-id="94343-113">詳細については、次を参照してください。 [XML および SOAP シリアル化](/dotnet/standard/serialization/xml-and-soap-serialization)します。</span><span class="sxs-lookup"><span data-stu-id="94343-113">For more information, see [XML and SOAP Serialization](/dotnet/standard/serialization/xml-and-soap-serialization).</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public InvalidPipelineStoreException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AddIn.Hosting.InvalidPipelineStoreException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; InvalidPipelineStoreException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.AddIn.Hosting.InvalidPipelineStoreException : string * Exception -&gt; System.AddIn.Hosting.InvalidPipelineStoreException" Usage="new System.AddIn.Hosting.InvalidPipelineStoreException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.AddIn</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="94343-114">エラーの説明。</span><span class="sxs-lookup"><span data-stu-id="94343-114">A description of the error.</span></span></param>
        <param name="innerException"><span data-ttu-id="94343-115">現在の例外の原因になった例外です。</span><span class="sxs-lookup"><span data-stu-id="94343-115">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="94343-116"><paramref name="innerException" /> パラメーターが null 参照でない場合は、内部例外を処理する catch ブロックで現在の例外が発生します。</span><span class="sxs-lookup"><span data-stu-id="94343-116">If the <paramref name="innerException" /> parameter is not a null reference, the current exception is raised in a catch block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="94343-117">指定したメッセージと、この例外の原因である内部例外への参照を使用して、<see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="94343-117">Initializes a new instance of the <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> class with the specified message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="94343-118">前回の例外の直接的な結果としてスローされる例外は、<xref:System.Exception.InnerException%2A> プロパティに前回の例外への参照を含める必要があります。</span><span class="sxs-lookup"><span data-stu-id="94343-118">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="94343-119"><xref:System.Exception.InnerException%2A> プロパティは、コンストラクターに渡されたものと同じ値を返します。 <xref:System.Exception.InnerException%2A> プロパティがコンストラクターに内部例外値を指定していない場合は、null 参照を返します。</span><span class="sxs-lookup"><span data-stu-id="94343-119">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed to the constructor, or a null reference if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="94343-120"><xref:System.AddIn.Hosting.AddInSegmentDirectoryNotFoundException> クラスのインスタンスの初期プロパティ値を次の表に示します。</span><span class="sxs-lookup"><span data-stu-id="94343-120">The following table shows the initial property values for an instance of the <xref:System.AddIn.Hosting.AddInSegmentDirectoryNotFoundException> class.</span></span>  
  
|<span data-ttu-id="94343-121">プロパティ</span><span class="sxs-lookup"><span data-stu-id="94343-121">Property</span></span>|<span data-ttu-id="94343-122">値</span><span class="sxs-lookup"><span data-stu-id="94343-122">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="94343-123">内部例外の参照です。</span><span class="sxs-lookup"><span data-stu-id="94343-123">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="94343-124">ローカライズされたエラー メッセージの文字列です。</span><span class="sxs-lookup"><span data-stu-id="94343-124">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>
